dts/arm/nxp/nxp_mcxn23x_common.dtsi

```diff
@@ -498,7 +498,7 @@
 		compatible = "nxp,mcux-edma";
```

Comment:
This PR LGTM, but it should be split in 2 commits:
commit 1 - adds description for dma-requests and dma-channels;
commit 2 - fixes dma-req values, which should also have a `Fixes` tag since those values were incorrect.